stl/inc/algorithm

```diff
@@ -785,6 +785,15 @@ namespace ranges {
         template <class _It1, class _Se1, class _It2, class _Se2, class _Pr, class _Pj1, class _Pj2>
         _NODISCARD static constexpr bool _Equal_4(
             _It1 _First1, _Se1 _Last1, _It2 _First2, _Se2 _Last2, _Pr _Pred, _Pj1 _Proj1, _Pj2 _Proj2) {
+            if constexpr (_Equal_memcmp_is_safe<_It1, _It2, _Pr> && same_as<_Pj1, identity> //

```

Comment:
I am quite sure that we do not need the full power of `same_as`

Would `is_same_v` or `_Same_impl` suffice?

Reply:
AFAIK most of the ranges algorithms use concepts rather than their `type_traits` counter parts.

Reply:
The point is that `same_as` checks for `_Same_impl<_Ty1, _Ty2> && _Same_impl<_Ty2,  _Ty1>` which seems abit too much here